### PR TITLE
consensus/local_snapshots: use base dir instead of base path

### LINKED ISSUE #1168: local_snapshots: use files base dir instead of base path

### PR DESCRIPTION
**Pull request by tsvisabo**
Fixes #1168

### CODE REVIEW

File: ciri/usage.h [hunk 1]
```diff
@@ -185,7 +185,7 @@ static struct cli_argument_s {
     {"local-snapshots-min-depth", CONF_LOCAL_SNAPSHOTS_MIN_DEPTH,
      "Minimal milestones depth for new local snapshot entry point.", REQUIRED_ARG},
     {"local-snapshots-path-base", CONF_LOCAL_SNAPSHOTS_PATH_BASE,
```
Comment (thibault-martinez) [inline ciri/usage.h]: Maybe we should also change the option name then ?

File: ciri/consensus/snapshot/snapshot.c [hunk 2]
```diff
@@ -163,15 +163,17 @@ retcode_t iota_snapshot_load_local_snapshot(snapshot_t *const snapshot, iota_con
   char file_path[256];
 
   strcpy(file_path, conf->local_snapshots.local_snapshots_path_base);
-  strcat(file_path, SNAPSHOT_METADATA_EXT);
+  strcat(file_path, iota_utils_file_separator());
```
Comment (thibault-martinez) [inline ciri/consensus/snapshot/snapshot.c]: `iota_utils_file_separator()` returns a `char`, not a `char *`

File: ciri/consensus/snapshot/snapshot.c [hunk 3]
```diff
@@ -163,15 +163,17 @@ retcode_t iota_snapshot_load_local_snapshot(snapshot_t *const snapshot, iota_con
...
+  strcat(file_path, SNAPSHOT_METADATA_FILE_NAME);
 
   if ((ret = (iota_snapshot_metadata_read_from_file(&snapshot->metadata, file_path)) != RC_OK)) {
     log_critical(logger_id, "Initializing snapshot metadata failed\n");
     return ret;
   }
 
   strcpy(file_path, conf->local_snapshots.local_snapshots_path_base);
-  strcat(file_path, SNAPSHOT_STATE_EXT);
+  strcat(file_path, iota_utils_file_separator());
```
Comment (thibault-martinez) [inline ciri/consensus/snapshot/snapshot.c]: `iota_utils_file_separator()` returns a `char`, not a `char *`

File: ciri/consensus/snapshot/snapshot.c (hunk 2 above)
Comment (thibault-martinez) [reply ciri/consensus/snapshot/snapshot.c]: I assume this is the reason tests are crashing with SEGV

File: ciri/consensus/snapshot/snapshot.c [hunk 4]
```diff
@@ -61,10 +61,10 @@ retcode_t iota_snapshot_write_to_file(snapshot_t const *const snapshot, char con
   }
 
   strcpy(state_path, snapshot_file_base);
```
Comment (thibault-martinez) [inline ciri/consensus/snapshot/snapshot.c:63]: Separator ?

File: ciri/consensus/snapshot/snapshot.c [hunk 5]
```diff
@@ -61,10 +61,10 @@ retcode_t iota_snapshot_write_to_file(snapshot_t const *const snapshot, char con
...
-  strcat(state_path, SNAPSHOT_STATE_EXT);
+  strcat(state_path, SNAPSHOT_STATE_FILE_NAME);
 
   strcpy(metadata_path, snapshot_file_base);
```
Comment (thibault-martinez) [inline ciri/consensus/snapshot/snapshot.c:67]: Separator ?

File: utils/files.h [hunk 6]
```diff
@@ -66,6 +66,21 @@ retcode_t iota_utils_read_file_into_buffer(char const *const file_path, char **c
...
 bool iota_utils_file_exist(char const *const file_path);
 
+/**
+ * Returns a char separator
+ *
+ * @param
+ *
+ * @return The char separator
+ */
+static inline char iota_utils_file_separator() {
```
Comment (thibault-martinez) [inline utils/files.h]: Maybe just make it a define ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: ciri/usage.h (hunk 1 above)
Comment (tsvisabo) [reply ciri/usage.h]: done

File: ciri/conf.c [hunk 7]
```diff
@@ -352,7 +352,7 @@ static retcode_t set_conf_value(iota_ciri_conf_t* const ciri_conf, iota_consensu
     case CONF_LOCAL_SNAPSHOTS_MIN_DEPTH:
       consensus_conf->local_snapshots.min_depth = atoi(value);
       break;
-    case CONF_LOCAL_SNAPSHOTS_PATH_BASE:
+    case CONF_LOCAL_SNAPSHOTS_BASE_DIR:
       strncpy(consensus_conf->local_snapshots.local_snapshots_path_base, value,
```
Comment (thibault-martinez) [inline ciri/conf.c]: Can you also rename the variable to `local_snapshots_base_dir` ?

Comment (thibault-martinez) [reply ciri/conf.c]: Or just `base_dir`, it's weird that it's repetitive since it's already in a `local_snapshots` struct.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Just a little nit

File: ciri/conf.c (hunk 7 above)
Comment (tsvisabo) [reply ciri/conf.c]: done

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**